bin/download_minister.py

```diff
@@ -0,0 +1,165 @@
+                    existing_list[row]['precedence'] = choices[row]['precedence']
+                    for langCode in lang_codes:
+                        existing_list[row][langCode] = choices[row][langCode]
+                        existing_list[row]['honorific_title_' + langCode] = choices[row]['honorific_title_' + langCode]
+                        # case: same minister with new position in choices, then append new position to existing list
+                        for position in choices[row]['positions_' + langCode]:
+                            if not [d for d in existing_list[row]['positions_' + langCode]
+                                    if d['title_' + langCode] == position['title_' + langCode]]:
+                                existing_list[row]['positions_' + langCode].append(position)
+                            else:
+                                # case: same minister with same position title but other details changed, then update
+                                idx = next((index for (index, d) in enumerate(existing_list[row]['positions_' + langCode]) if d['title_' + langCode] == position['title_' + langCode]), None)
+                                existing_list[row]['positions_' + langCode][idx].update(position)
+                        # case: same minister with existing position not in choices then update end date as today
+                        for position in existing_list[row]['positions_' + langCode]:
+                            if not [d for d in choices[row]['positions_' + langCode]
+                                    if d['title_' + langCode] == position['title_' + langCode]]:
+                                position['end_date'] = date.today().strftime("%Y-%m-%dT%H:%M:%S")
+            else:
+                # case: minister found in existing list but now in new list, update end date as today
+                for langCode in lang_codes:
+                    for position in existing_list[row]['positions_' + langCode]:
+                        position['end_date'] = date.today().strftime("%Y-%m-%dT%H:%M:%S")
+
+        for row in choices:
+            if row not in existing_list:
+                # case: new minister in choices, add new minister to existing list
+                existing_list[row] = choices[row]
+
+        return set_status(existing_list)
+    # both lists do not exist
+    else:
+        return False
+
+
+def download_xml_from_source():
+    """
+    Read XML from external source to generate JSON choices for ministers
+    """
+    choices = {}
+    xml_tree = {}
+
+    # collect responses for both languages
+    for langCode in lang_codes:
+        # url = 'https://www.ourcommons.ca/Parliamentarians/' + langCode + '/ministries/Export?output=XML'
+        url = 'http://localhost/export-' + langCode + '.xml'
+        response = requests.get(url, stream=True)
+
+        # if valid response then save as xml tree
+        if response.status_code == 200:
```

Comment:
requests will raise an exception on an error, so there's no need to check the status code.